src/MessagePack.Annotations/Attributes.cs

```diff
@@ -124,6 +128,9 @@ public class SerializationConstructorAttribute : Attribute
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface | AttributeTargets.Enum | AttributeTargets.Field | AttributeTargets.Property | AttributeTargets.Parameter | AttributeTargets.ReturnValue, AllowMultiple = false, Inherited = true)]
     public class MessagePackFormatterAttribute : Attribute
     {
+#if NET8_0_OR_GREATER
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields | DynamicallyAccessedMemberTypes.Interfaces)]
```

Comment:
I think you also need this attribute on the `Type` parameters passed into the constructors.

Reply:
That sounds reasonable. I wonder why the analyzer missed that.